src/Microsoft.Azure.SignalR.Common/ServiceConnections/ServiceConnectionBase.cs

```diff
@@ -430,7 +480,14 @@ private async Task ProcessIncomingAsync(ConnectionContext connection)
 
                             while (ServiceProtocol.TryParseMessage(ref buffer, out var message))
                             {
-                                _ = DispatchMessageAsync(message);
+                                if (message is CloseConnectionMessage closeConnectionMessage && closeConnectionMessage.ConnectionId == Constants.ConnectionFinAck)
+                                {
+                                    inLoop = false;

```

Comment:
`return`?

Reply:
I'm not sure what else should be done after the while loop so I can't simply return here.